boskos/mason/mason.go

```diff
@@ -286,32 +313,23 @@ func (m *Mason) freeAll(ctx context.Context) {
 }
 
 func (m *Mason) freeOne(res *common.Resource) error {
+	leasedResources, err := checkUserData(*res)
+	if err != nil {
+		return err
+	}
+	// TODO: Implement a ReleaseMultiple in a transaction to prevent orphans
 	// Finally return the resource as free
 	if err := m.client.ReleaseOne(res.Name, common.Free); err != nil {
 		logrus.WithError(err).Errorf("failed to release resource %s", res.Name)
 		return err
 	}
-	var leasedResources common.LeasedResources
-	if res.UserData == nil {
-		err := fmt.Errorf("UserData is empty")
-		logrus.WithError(err).Errorf("failed to extract %s", LeasedResources)
-		return err
-	}
-	if err := res.UserData.Extract(LeasedResources, &leasedResources); err != nil {
-		logrus.WithError(err).Errorf("failed to extract %s", LeasedResources)
-		return err
-	}
 	// And release leased resources as res.Name state
-	var allErrors error
 	for _, name := range leasedResources {
 		if err := m.client.ReleaseOne(name, res.Name); err != nil {
 			logrus.WithError(err).Errorf("unable to release %s to state %s", name, res.Name)
-			allErrors = multierror.Append(allErrors, err)
+			return err
```

Comment:
why? how about other leased resources?

Reply:
So I changed this recently and this was a mistake. So if one the resource cannot be released, there is no point if releasing the other one. In freeAll they'll be all release as dirty (or at least the remaining one). In order to properly handle that case, I need to implement this on Ranch direclty.